tests/integration/integrations/conftest.py

```diff
@@ -5,61 +5,206 @@
 
 import collections
 import logging
-import os
 import pathlib
 import time
 
+import jubilant
 import kubernetes
-import nest_asyncio
 import pytest
-import pytest_asyncio
-from juju.application import Application
 from minio import Minio
-from ops import JujuVersion
-from pytest_operator.plugin import OpsTest
+
+from tests.integration.conftest import build_charm_file
+from tests.integration.types import App
 
 PROJECT_ROOT = pathlib.Path(__file__).parent.parent.parent.parent
-nest_asyncio.apply()
 
 logger = logging.getLogger(__name__)
 
 
-@pytest.fixture(autouse=True)
-def skip_by_juju_version(request, model):
-    """Skip the test if juju version is lower then the `skip_juju_version` marker value."""
-    if request.node.get_closest_marker("skip_juju_version"):
-        current_version = JujuVersion(
-            f"{model.info.agent_version.major}.{model.info.agent_version.minor}.{model.info.agent_version.patch}"
+def deploy_postgresql(
+    juju: jubilant.Juju,
+):
+    """Deploy and set up postgresql charm needed for the 12-factor charm."""
+
+    if juju.status().apps.get("postgresql-k8s"):
+        logger.info("postgresql-k8s already deployed")
+        return
+
+    juju.deploy(
+        "postgresql-k8s",
+        channel="14/stable",
+        base="ubuntu@22.04",
+        revision=300,
+        trust=True,
+        config={"profile": "testing"},
+    )
```

Comment:
```suggestion
    juju.deploy(
        "postgresql-k8s",
        channel="14/stable",
        base="ubuntu@22.04",
        revision=300,
        trust=True,
        config={
        "profile": "testing",
        "plugin_hstore_enable": "true",
        "plugin_pg_trgm_enable": "true",
        },
    )
```

Could this help reduce the double wait? (I'm not sure what the behavior of it is, perhaps it is intended to apply configuration change afterwards).